src/lightgbm/src/main/scala/LightGBMBooster.scala

```diff
@@ -138,4 +139,13 @@ class LightGBMBooster(val model: String) extends Serializable {
     }
     new LightGBMBooster(model)
   }
+
+  def saveModel(session: SparkSession, filename: String): Unit = {
+    if (filename == null || filename.isEmpty()) {
+      throw new Exception("filename should not be empty or null.")
+    }
+
+    val rdd = session.sparkContext.parallelize(Seq(model))
+    rdd.coalesce(1).saveAsTextFile(filename)
```

Comment:
Are you sure this is the format LightGBM native expects?

Reply:
Actually, I just realize there are some differences at the end of the file:

The model saved by Spark is the following:

```
feature importances:
Column_22=573
Column_20=374
Column_21=151
Column_9=119
Column_14=59
Column_3=57
Column_7=48
Column_11=44
Column_1=23
Column_2=13
Column_6=9
Column_13=7
Column_12=6
Column_4=6
Column_8=5
Column_5=4
Column_10=2
```

The model saved from native lightgbm:

```
feature importances:
HourOfDay=3009
DaysSinceOobe=2433
UniqueAppsInFocus600_Clipped=870
UserInputA600_Clipped=759
DayOfWeek=155
MouseInputA600_Clipped=74
UniqueAppsInFocus60_Clipped=66
KeyboardInputA600_Clipped=49
UserInputToUniqueAppsRatio60=32
MouseInputA60_Clipped=18
UserInputA60_Clipped=17
UserInputVelocity=10
UniqueAppsInFocusVelocity=5
MouseToKeyboardRatio60=3

pandas_categorical:[]
```

I think we will need to address this tomorrow.